underscore.js

```diff
@@ -1340,8 +1347,16 @@
 
   // Shortcut function for checking if an object has a given property directly
   // on itself (in other words, not on a prototype).
-  _.has = function(obj, key) {
-    return obj != null && hasOwnProperty.call(obj, key);
+  _.has = function(object, path) {
+    if (!_.isArray(path)) path = [path];
```

Comment:
For this and property I wonder if it'd be better to fast track it (i.e. use the old implementation)